File: src/online/activity.ts
import { Column, Entity, ObjectIdColumn } from 'typeorm';

export class Activity {
  pilot: number;
  atc: number;
}

@Entity('activity')
export class NetworksActivity {
  @ObjectIdColumn()
  id: string;

  @Column()
  date: Date;

  @Column()
  ivao: Activity;

  @Column()
  vatsim: Activity;

  @Column()
  poscon: Activity;
}

@Entity('br_activity')
export class BrazilNetworksActivity {
  @ObjectIdColumn()
  id: string;

  @Column()
  date: Date;

  @Column()
  ivao: Activity;

  @Column()
  vatsim: Activity;

  @Column()
  poscon: Activity;
}

export class GeoActivity {
  [fir: string]: Activity;
}
@Entity('geo_activity')
export class GeoNetworksActivity {
  @ObjectIdColumn()
  id: string;

  @Column()
  date: Date;

  @Column()
  ivao: GeoActivity;

  @Column()
  vatsim: GeoActivity;

  @Column()
  poscon: GeoActivity;
}

File: internal/domain/activity.go
package domain

import (
	"time"
)

type Activity struct {
	Pilot int64 `json:"pilot"`
	ATC   int64 `json:"atc"`
}

type GeoActivity map[string]Activity

type NetworkActivity struct {
	//ID     primitive.ObjectID `bson:"_id" json:"id,omitempty"`
	Date   time.Time `json:"date"`
	IVAO   Activity  `json:"ivao"`
	VATSIM Activity  `json:"vatsim"`
	POSCON Activity  `json:"poscon"`
}

type GeoNetworkActivity struct {
	//ID     primitive.ObjectID `bson:"_id" json:"id,omitempty"`
	Date   time.Time   `json:"date"`
	IVAO   GeoActivity `json:"ivao"`
	VATSIM GeoActivity `json:"vatsim"`
	POSCON GeoActivity `json:"poscon"`
}
